firebase-firestore/src/test/java/com/google/firebase/firestore/spec/SpecTestCase.java

```diff
@@ -1006,6 +1006,9 @@ private void validateExpectedState(@Nullable JSONObject expectedState) throws JS
                   targetData.withResumeToken(
                       ByteString.EMPTY, version(queryDataJson.getInt("readTime")));
             }
+            if (queryDataJson.has("expectedCount")) {
```

Comment:
Wow, is this the only change you had to make to the spec test runner? Can you just confirm that if you modify the spec tests so that they should fail that they do actually fail? I'm sure you tried this, but I'm just surprised that updating the spec tests was so easy!

Reply:
Yes, in WEB we updated the spec_test_builder and spec test file as well. But in Android, we just need to wait for WEB to "build spec test"->"serialize into json file" -> "import into android" -> "deserialize and run spec test".

so only need to update spec test runner.

Reply:
this Spec is only to add the expectedCount. In the next PR, for applying bloom filter, we need to do much more than this.